src/toil/job.py

```diff
@@ -2450,35 +2457,31 @@ class PromisedRequirementFunctionWrappingJob(FunctionWrappingJob):
     """
     Handles dynamic resource allocation using :class:`toil.job.Promise` instances.
     Spawns child function using parent function parameters and fulfilled promised
-    resource requirements. See :class:`toil.job.FunctionWrappingJob` class
+    resource requirements. This job must be encapsulated (see :func:`toil.job.encapsulate`)
+    to avoid a race condition between child functions.
     """
     def __init__(self, userFunction, *args, **kwargs):
-        self.promisedRequirements = {}
-        requirements = {'disk': '1M', 'memory': '32M', 'cores': 0.1}
-        # Replace PromisedRequirements in intermediate job with small
-        # resource requirements.
-        for name, value in requirements.items():
-            try:
-                if isinstance(kwargs[name], PromisedRequirement):
-                    self.promisedRequirements[name] = kwargs[name]
-                    kwargs[name] = value
-            except KeyError:
-                pass
+        self._promisedKwargs = kwargs.copy()
+        minRequirements = {'disk': '1M', 'memory': '32M', 'cores': 0.1}
+        # Replace resource requirements in intermediate job with minimum values.
```

Comment:
Minimal, not minimum. Also, they are not minimal either, their just small. I would cut out the whole "min" thing and not even have a special variable for this. `kwargs.update( dict( disk=…, memory=… ) )` is very obvious.

Yes, and learn about the `dict` constructor as an alternative to dictionary literals.